samcli/lib/delete/cfn_utils.py

```diff
@@ -56,27 +71,56 @@ def has_stack(self, stack_name: str) -> bool:
             LOG.error("Botocore Exception : %s", str(e))
             raise DeleteFailedError(stack_name=stack_name, msg=str(e)) from e
 
-    def get_stack_template(self, stack_name: str, stage: str) -> Dict:
+    def get_stack_template(self, stack_name: str, stage: str) -> str:
         """
         Return the Cloudformation template of the given stack_name
 
-        :param stack_name: Name or ID of the stack
-        :param stage: The Stage of the template Original or Processed
-        :return: Template body of the stack
+        Parameters
+        ----------
+
+        stack_name: str
+            Name or ID of the stack
+        stage: str
+            The Stage of the template Original or Processed
+
+        Returns
+        -------
+        str
+            Template body of the stack
+
+        Raises
+        ------
+        FetchTemplateFailedError
+            Raised when boto calls or parsing fails to fetch template
         """
         try:
             resp = self._client.get_template(StackName=stack_name, TemplateStage=stage)
-            if not resp["TemplateBody"]:
-                return {}
-            return dict(resp)
+            template = resp.get("TemplateBody", "")
+
+            # stack may not have template, check the change set
+            if not template:
+                change_set_name = self._get_change_set_name(stack_name)
+
+                if change_set_name:
+                    # the stack has a change set, use the template from this
+                    resp = self._client.get_template(
+                        StackName=stack_name, TemplateStage=stage, ChangeSetName=change_set_name
+                    )
+                    template = resp.get("TemplateBody", "")
+
+            return str(template)
 
         except (ClientError, BotoCoreError) as e:
             # If there are credentials, environment errors,
             # catch that and throw a delete failed error.
 
             LOG.error("Failed to fetch template for the stack : %s", str(e))
             raise FetchTemplateFailedError(stack_name=stack_name, msg=str(e)) from e
-
+        except FetchChangeSetError as ex:
+            raise FetchTemplateFailedError(stack_name=stack_name, msg=str(ex)) from ex
+        except NoChangeSetFoundError as ex:
+            msg = f"Failed to find a change set for stack {stack_name} to fetch the template"
```

Comment:
The stack name part in the msg might seem a bit redundant as the exception will also use the `stack_name` attr in formatting the final message.

Reply:
Good catch, will remove the dup stack name